<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Yifan Ji" />


<title>Introduction to contamDE</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#header {
text-align: center;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; }  code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>

</head>

<body>




<h1 class="title toc-ignore">Introduction to contamDE</h1>
<h4 class="author"><em>Yifan Ji</em></h4>
<h4 class="date"><em>2019/5/7</em></h4>



<div id="method-description" class="section level1">
<h1>Method description</h1>
<p>The R package <strong>contamDE</strong> implements two robust and powerful statistical methods for differential expression analysis using next generation RNA-seq data from contaminated tumor samples.</p>
<div id="description-and-illustration-of-the-first-method" class="section level2">
<h2>Description and illustration of the first method</h2>
<p>The method is able to deal with both contaminated matched data with a single or multiple tumor cell types and unmatched data by estimating the purity proportions as well as dispersion parameters based on negative binomial model. The data for each normal-tumor matched samples are <span class="math inline">\((N_{ij},T_{ij},e^{\alpha_{ij}})\)</span>, where <span class="math inline">\(N_{ij}\)</span> and <span class="math inline">\(T_{ij}\)</span> are the number of the RNA-seq reads mapped to the <span class="math inline">\(j\)</span>th gene for the <span class="math inline">\(i\)</span>th normal and tumor cell sample respectively, and <span class="math inline">\(\alpha_{ij}\)</span> is one common fix effect to characterize the correlation between matched samples. Suppose tumor cell sample consists of a proportion of <span class="math inline">\(w_i\)</span> tumor cells and a proportion of <span class="math inline">\(1-w_i\)</span> normal cells, then the model for contaminated and matched samples with a common dispersion <span class="math inline">\(\phi_j\)</span> and mean parameter of normal samples <span class="math inline">\(\mu_j\)</span> is as follows: <span class="math display">\[\begin{eqnarray*}
&amp;&amp;N_{ij} \sim NB(\kappa_i\mu_je^{\alpha_{ij}},\phi_j)\\
&amp;&amp;T_{ij} \sim NB(\kappa&#39;_i(\mu_j+w_i\delta_j)e^{\alpha_{ij}},\phi_j)
\end{eqnarray*}\]</span> where <span class="math inline">\(\kappa_i\)</span> and <span class="math inline">\(\kappa&#39;_i\)</span> are size factors for normal and tumor samples; <span class="math inline">\(\delta_j\)</span> is the mean expression difference between tumors and normals which corresponds to <span class="math inline">\(\delta_j=\mu&#39;_j-\mu_j\)</span> respectively.</p>
<p>The purity proportions <span class="math inline">\(w_i\)</span> are identifiable up to a scale parameter. After constructing an initial estimator of <span class="math inline">\(w_i\)</span>, we update the estimator based on a likelihood method. The initial estimator of <span class="math inline">\(w_i\)</span> given by <span class="math display">\[\begin{eqnarray*}
&amp;&amp;\hat{w_i}^{(1)}=\frac{I\sum_{j\in\Omega_1}(T_{ij}/\kappa&#39;_i-N_{ij}/\kappa_i)}{\sum_{i=1}^I\sum_{j\in\Omega_1}(T_{ij}/\kappa&#39;_i-N_{ij}/\kappa_i)}
\end{eqnarray*}\]</span> In practice, we can choose the working gene set <span class="math inline">\(\Omega_1\)</span> using any DE analysis tool such as edgeR. More accurate estimators of <span class="math inline">\(w_i\)</span> can be iteratively given by <span class="math display">\[\begin{eqnarray*}
&amp;&amp;\hat{w_i}^{(k)}=\frac{I\tilde{w_i}}{\sum_{i&#39;=1}^I\tilde{w_{i&#39;}}}\mbox{with} \ \tilde{w_i}=\frac{\sum_{j\in\Omega_k}(T_{ij}/\kappa&#39;_i-N_{ij}/\kappa_i)}{I^{-1}\sum_{i=1}^I\sum_{j\in\Omega_k}\hat{\delta}^{(k-1)}_je^{\hat{\alpha}_{ij}^{(k-1)}}}
\end{eqnarray*}\]</span> for <span class="math inline">\(k=2,3,\cdots\)</span> The iteration can be stopped if <span class="math inline">\(max_i|\hat{w_i}^{(k+1)}-\hat{w_i}^{(k)}| \le \epsilon\)</span> for some small <span class="math inline">\(\epsilon\)</span>.</p>
<p>With given purity proportion estimates <span class="math inline">\(w(k)=(\hat{w}_1^{(k)},\cdots,\hat{w}_I^{(k)}),k=1,2,\cdots,\)</span> and fixed <span class="math inline">\(\hat{\phi}_j\)</span>, we estimate <span class="math inline">\(\mu_j\)</span>, <span class="math inline">\(\delta_j\)</span> and <span class="math inline">\(\alpha_j\)</span> by maximizing a log-pseudo likelihood function. The pseudo-likelihood ratio test statistics for testing null hypotheses <span class="math inline">\(H_j:\delta_j=0\)</span> can be then constructed and the fold change can be estimated by <span class="math display">\[\begin{eqnarray*}
&amp;&amp; FC_j^{(k)}=\hat{\delta}_j^{(k)}/\hat{\mu}_j^{(k)}+1
\end{eqnarray*}\]</span></p>
The usage of the functions <em>contamDE</em> is as follows:

<p>Users should specify <em>data</em>, <em>R</em>, <em>n</em>, <em>match</em> according to their requirements. <em>data</em> is a <span class="math inline">\(G*N\)</span> read count matrix, where <span class="math inline">\(G\)</span> is the number of genes and <span class="math inline">\(N\)</span> is the total sample size. Conditions where to detect difference on gene expression can be designed by specifying <em>R</em> (the number of conditions) and <em>n</em> (a list of sample sizes for all conditions and will be valid if data is unmatched). If data are samples in pair, set <em>match</em> to be <em>TRUE</em>.</p>
</div>
</div>
<div id="illustration-of-contamde" class="section level1">
<h1>Illustration of <em>contamDE</em></h1>
<div id="matched-samples" class="section level2">
<h2>Matched samples</h2>
<p>We use dataset ‘prostate’ contained in the R package for illustration.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1"><span class="kw">library</span>(contamDE)</a></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" title="1"><span class="kw">data</span>(<span class="st">&quot;prostate&quot;</span>,<span class="dt">package=</span><span class="st">&quot;contamDE&quot;</span>)</a>
<a class="sourceLine" id="cb2-2" title="2"><span class="kw">dim</span>(prostate)</a></code></pre></div>
<pre><code>## [1] 12698    29</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" title="1"><span class="kw">head</span>(prostate)</a></code></pre></div>
<pre><code>##   nameOfGene N1 N2 N3 N4 N5 N6 N7 N8 N9 N10 N11 N12 N13 N14 T1 T2 T3 T4 T5
## 1      KRT36  1 13  4  0 51  1  3  0  3   0   2   0  13   1  0  2  0  0  0
## 2  KRTAP13-2  2  0  0  0  0  5  3  0 68   0   0   7   0   1  0  0  2  0  0
## 3       LGR5  3  1 11  3  0  4  0  2  0   0  12  11   0  64  4  2  5  0  0
## 4      SMR3B  7  0  0  0  0  0  0  0  0  57   0  40   0   0 26  0  0  1  0
## 5       CST4  3  4 62  8  0  2  1  5 17  15   1   6   2   6  1  0  0  3  1
## 6     RNF186  1  3 10  0 53  1  1  4  0   0  20  12  20  11  0  1  1  1  1
##   T6 T7 T8 T9 T10 T11 T12 T13 T14
## 1  1  0  0  0   0   0   0   0   0
## 2  4  3  0  0   0   1   1   0   0
## 3  2  1  0  1   0   0   2   2   2
## 4  1  0  0  0   0   0   0   0   0
## 5  1  4  2  0   3   0   3   0   5
## 6  4  0  1  0   0   5   4   5   0</code></pre>
<p>This dataset contains the read count of 12699 genes for prostate cancer samples and adjacent normal cells samples which provided by 14 patients from Shanghai Changhai Hospital. Suppose we are interested in the differential expression of genes between two matched samples.</p>
<p>The R code for analyzing the data is as follows:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" title="1"><span class="kw">library</span>(<span class="st">&quot;contamDE&quot;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" title="1">d &lt;-<span class="st"> </span><span class="kw">contamDE</span>(prostate[,<span class="op">-</span><span class="dv">1</span>],<span class="dt">R=</span><span class="dv">2</span>,<span class="dt">match=</span><span class="ot">TRUE</span>)</a></code></pre></div>
<p>The result is a list containing the following elements:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" title="1"><span class="kw">names</span>(d)</a></code></pre></div>
<pre><code>## [1] &quot;W&quot;  &quot;LR&quot;</code></pre>
<p>The purity proportion estimates for each pair produced by the new method are</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" title="1">d<span class="op">$</span>W</a></code></pre></div>
<pre><code>##  [1] 1.3078678 0.8623965 1.0131880 1.0174180 1.2064407 0.5894919 1.1149573
##  [8] 0.9509469 1.1295711 1.2578447 0.7663290 0.6022404 1.2513781 0.9299297</code></pre>
<p>The pseudo likelihood test results by the new method are</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" title="1"><span class="kw">head</span>(d<span class="op">$</span>LR)</a></code></pre></div>
<pre><code>##             lrt      p.value      logFC
## [1,] 21.8686372 2.919670e-06 -1.7268079
## [2,]  2.4070001 1.207936e-01 -0.8302163
## [3,]  2.7050280 1.000323e-01 -0.6706763
## [4,]  0.1693806 6.806629e-01 -0.3048375
## [5,]  9.3975735 2.172728e-03 -1.2926252
## [6,] 16.7616778 4.238062e-05 -1.5243291</code></pre>
</div>
<div id="unmatched-samples" class="section level2">
<h2>Unmatched samples</h2>
<p>We use dataset ‘drosophila’ contained in the R package for illustration.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" title="1"><span class="kw">data</span>(<span class="st">&quot;drosophila&quot;</span>,<span class="dt">package=</span><span class="st">&quot;contamDE&quot;</span>)</a>
<a class="sourceLine" id="cb14-2" title="2"><span class="kw">dim</span>(drosophila)</a></code></pre></div>
<pre><code>## [1] 7196    8</code></pre>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" title="1"><span class="kw">head</span>(drosophila)</a></code></pre></div>
<pre><code>##    nameOfGene CT.PA.1 CT.PA.2 CT.SI.5 CT.SI.7 KD.PA.3 KD.SI.6 KD.PA.4
## 1 FBgn0000008      76      71     137      82      87      68     115
## 2 FBgn0000017    3498    3087    7014    3926    3029    3264    4322
## 3 FBgn0000018     240     306     613     485     288     307     528
## 4 FBgn0000032     611     672    1479    1351     694     757    1361
## 5 FBgn0000042   40048   49144   97565   99372   70574   72850   95760
## 6 FBgn0000043   15910   18194   34171   29671   31086   34085   42389</code></pre>
<p>This dataset consists of the RNA-seq read counts of 7196 genes for 7 unmatched samples of Drosphila melanogaster S2 cells, of which 4 samples were untreated while 3 samples were treated. The R code for analyzing the data is as follows:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" title="1"><span class="kw">library</span>(<span class="st">&quot;contamDE&quot;</span>)</a></code></pre></div>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb19-1" title="1">d &lt;-<span class="st"> </span><span class="kw">contamDE</span>(drosophila[,<span class="op">-</span><span class="dv">1</span>],<span class="dt">R=</span><span class="dv">2</span>,<span class="dt">n=</span><span class="kw">list</span>(<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>,<span class="dv">5</span><span class="op">:</span><span class="dv">7</span>),<span class="dt">match=</span><span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb19-2" title="2">d<span class="op">$</span>W</a></code></pre></div>
<pre><code>## [1] 0.9743470 0.9818007 1.0438522</code></pre>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb21-1" title="1"><span class="kw">head</span>(d<span class="op">$</span>LR)</a></code></pre></div>
<pre><code>##               lrt      p.value        logFC
## [1,] 2.177926e-05 0.9962764259  0.001030782
## [2,] 1.959686e+00 0.1615468840 -0.256221146
## [3,] 2.335351e-01 0.6289148683 -0.085454003
## [4,] 2.098455e-01 0.6468885017 -0.087014070
## [5,] 1.495160e+00 0.2214175509  0.242913757
## [6,] 1.284316e+01 0.0003387145  0.618098721</code></pre>
</div>
<div id="description-and-illustration-of-the-second-method" class="section level2">
<h2>Description and illustration of the second method</h2>
<p>The method is similar with the first method since it mainly deal with RNA-seq data from contaminated tumor samples. This DE analysis procedure is based on a novel linear model and allows for pair-specific residual variances. Covariates can also be incorporated in it. Compared with the first method under normal-tumor matched condition, it is computationally much faster and more statistically robust, especially when the pair number is large.</p>
<p>The data for each normal-tumor matched samples are <span class="math inline">\((N_{ig},T_{ig})\)</span>, where <span class="math inline">\(N_{ig}\)</span> and <span class="math inline">\(T_{ig}\)</span> are the read counts at the <span class="math inline">\(g\)</span>th gene for the <span class="math inline">\(i\)</span>th normal and tumor cell sample respectively. The novel linear model using a transformed read count as the response can be formed as: <span class="math display">\[\begin{eqnarray*}
&amp;&amp; y_{ig}=log(\frac{T_{ig}}{\kappa_{iT}}+1)-log(\frac{N_{ig}}{\kappa_{iN}}+1)=w_i(x_i^T\beta_g+z_i^T\gamma_g)+e_{ig}
\end{eqnarray*}\]</span> where <span class="math inline">\(y_{ig}\)</span> is the difference between two conditions; <span class="math inline">\(\kappa_{iT}\)</span> and <span class="math inline">\(\kappa_{iN}\)</span> are respectively the size factor of tumor and normal samples; <span class="math inline">\(x_i\)</span> indicates tumor subtype and <span class="math inline">\(z_i\)</span> is a q-vector of covariates; <span class="math inline">\(\beta_g\)</span> is the log-fold change between tumor tissues and <span class="math inline">\(\gamma_g\)</span> accounts for covariates. Note that <span class="math inline">\(w_i\)</span> are still purity proportions and identifiable up to a scale parameter.</p>
<p>With estimated parameter <span class="math inline">\(w_i\)</span>, <span class="math inline">\(\kappa_{iT}\)</span>,<span class="math inline">\(\kappa_{iN}\)</span>,<span class="math inline">\(\theta_g=(\beta_g^T,\gamma_g^T)^T\)</span> and the variance-covariance matrix of <span class="math inline">\(\theta_g\)</span>, either the t-test (for two conditions) or F-test (for at least three conditions) can be conducted by specifying a non-degenerated matrix：<span class="math inline">\(H\)</span>. To perform a DE analysis, the null hypothesis can take a form of <span class="math inline">\(H_0:H\beta_g=b\)</span>. For example, when detecting differentially expressed genes (DEGs) between the tumor and the normal, we ignore the subtype information (if any) and the hypothesis test problem can be formulated as <span class="math inline">\(H_0:\beta_g=0\)</span> which corresponds to <span class="math inline">\(H = 1\)</span>；when detecting DEGs between tumor subtype 1 and tumor subtype 2, we formulate the null hypothesis as <span class="math inline">\(H_0:\beta_{g1}=\beta_{g2}\)</span> which corresponds to <span class="math inline">\(H=(1,-1,0,\cdots,0)\)</span> and when it comes to detect DEGs between any tumor subtype and the normal, the null hypothesis can be <span class="math inline">\(H_0:\beta_{g1}=\cdots=\beta_{gp}=0\)</span> where <span class="math inline">\(p\)</span> is the subtype number. Note that any two-sided t-test is equivalent to a F-test with the first degree of freedom being one. The corresponding F-test statistic is <span class="math display">\[\begin{eqnarray*}
&amp;&amp; F_g=(H\hat{\theta}_g-b)^T(H\hat{\Sigma}_gH^T)^{-1}(H\hat{\theta}_g-b)/r
\end{eqnarray*}\]</span> which approximately follows the F-distribution with degrees of freedom <span class="math inline">\(r\)</span> and <span class="math inline">\(d_0+d_g\)</span>.</p>
The usage of the functions <em>contamDE.lm</em> is as follows:

<p>Users should specify <em>counts</em>, <em>subtype</em>, <em>covariate</em>, <em>is.contaminated</em> and <em>robust</em> according to their requirements. <em>counts</em> is a <span class="math inline">\(G*N\)</span> read count matrix, where <span class="math inline">\(G\)</span> is the number of genes and <span class="math inline">\(N\)</span> is the total sample size. <em>subtype</em> and <em>covariate</em> can be specified in the model by setting it to be the default value NULL. If data don’t have contamination or outliers, both <em>is.contaminated</em> and <em>robust</em> can be setted to be <em>FALSE</em>.</p>
</div>
</div>
<div id="illustration-of-contamde.lm" class="section level1">
<h1>Illustration of <em>contamDE.lm</em></h1>
<p>Use dataset ‘mixcounts’ contained in the R package for illustration.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" title="1"><span class="kw">library</span>(<span class="st">&quot;contamDE&quot;</span>)</a>
<a class="sourceLine" id="cb23-2" title="2"><span class="kw">data</span>(<span class="st">&quot;simcounts&quot;</span>)</a>
<a class="sourceLine" id="cb23-3" title="3"><span class="kw">dim</span>(simcounts)</a></code></pre></div>
<pre><code>## [1] 10000    16</code></pre>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" title="1"><span class="kw">head</span>(simcounts)</a></code></pre></div>
<pre><code>##   Type1Normal1 Type1Normal2 Type1Normal3 Type1Normal4 Type2Normal1
## 1         2249         1465         2564         1163         1408
## 2         3649         4165        11399         3788         6075
## 3          316          281          263          475         1195
## 4         2030         5774         5292         3297         3864
## 5           91           92          299          133          160
## 6           67           19          119           14          114
##   Type2Normal2 Type3Normal1 Type3Normal2 Type1Tumor1 Type1Tumor2
## 1         1210          860         1611        1407        1531
## 2         3528         1822         7375        3418        4267
## 3          370          408         1367          69         164
## 4         4951          794         8461        3030        6941
## 5          120           93          164          65         110
## 6           73           43           95         102          16
##   Type1Tumor3 Type1Tumor4 Type2Tumor1 Type2Tumor2 Type3Tumor1 Type3Tumor2
## 1        1231        4295        2808        1417        2726        4198
## 2        3557       11479       14604        5573        8310       15949
## 3         117        3138        2808         666          89        1726
## 4        5570       38820       18996        3359        7410       21441
## 5         149         469         432         110         185         492
## 6          53         377         128          91          42         295</code></pre>
<p>The dataset contains simulated RNA-Seq read counts for eight patients with both normal and tumor samples. The eight patients belong to three tumor subtypes, i.e., four ‘Type1’ patients, two ‘Type2’ patients, and two ‘Type3’ patients.</p>
<div id="without-subtypes" class="section level2">
<h2>Without subtypes</h2>
<p>Suppose we are only interested in difference between the contaminated tumor and normal samples, then a t-test can be carried out by specifying the subtype and covariates information through <em>subtype</em> and <em>covariate</em>:</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb27-1" title="1">subtype &lt;-<span class="st"> </span><span class="ot">NULL</span></a>
<a class="sourceLine" id="cb27-2" title="2">d &lt;-<span class="st"> </span><span class="kw">contamDE.lm</span>(simcounts,subtype)</a></code></pre></div>
<p>The result is a list containing the following elements:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb28-1" title="1"><span class="kw">names</span>(d)</a></code></pre></div>
<pre><code>## [1] &quot;counts&quot;        &quot;p.contamDE.lm&quot; &quot;log2FC&quot;        &quot;log2FC.cov&quot;   
## [5] &quot;proportion&quot;    &quot;design&quot;        &quot;df&quot;            &quot;weight&quot;       
## [9] &quot;y&quot;</code></pre>
<p>The corresponding statistical results of t-test by this method are</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb30-1" title="1"><span class="kw">head</span>(d<span class="op">$</span>p.contamDE.lm)</a></code></pre></div>
<pre><code>##      subtype-1 vs. normal
## [1,]         2.565351e-03
## [2,]         6.180637e-03
## [3,]         8.978552e-01
## [4,]         7.006808e-05
## [5,]         2.158559e-03
## [6,]         2.706327e-02</code></pre>
<p>The purity proportions of each sample are</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb32-1" title="1">d<span class="op">$</span>proportion</a></code></pre></div>
<pre><code>## Type1Tumor1 Type1Tumor2 Type1Tumor3 Type1Tumor4 Type2Tumor1 Type2Tumor2 
##   0.4067528   0.8861346   0.5380201   0.9639193   1.0000000   0.1542927 
## Type3Tumor1 Type3Tumor2 
##   0.6416266   0.9474743</code></pre>
<p>The <span class="math inline">\(log_2FC\)</span> (the logarithm value of fold changes) between two conditions (tumor versus normal) are</p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb34-1" title="1"><span class="kw">head</span>(d<span class="op">$</span>log2FC)</a></code></pre></div>
<pre><code>##            [,1]
## [1,] 0.84224717
## [2,] 0.84963348
## [3,] 0.07720014
## [4,] 1.99192680
## [5,] 0.93070709
## [6,] 1.34051789</code></pre>
<p>The design matrix of the new method under two condition is</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb36-1" title="1">d<span class="op">$</span>design</a></code></pre></div>
<pre><code>##      [,1]
## [1,]    1
## [2,]    1
## [3,]    1
## [4,]    1
## [5,]    1
## [6,]    1
## [7,]    1
## [8,]    1</code></pre>
</div>
<div id="incorporate-subtypes" class="section level2">
<h2>Incorporate subtypes</h2>
<p>Suppose we are now interested in difference between any subtype and normal samples (e.g. subtype1(2/3) versus normal), then the R code of conducting a t-test incorporating subtype information are</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb38-1" title="1">subtype &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">rep</span>(<span class="st">&quot;Type1&quot;</span>,<span class="dv">4</span>),<span class="kw">rep</span>(<span class="st">&quot;Type2&quot;</span>,<span class="dv">2</span>),<span class="kw">rep</span>(<span class="st">&quot;Type3&quot;</span>,<span class="dv">2</span>))</a>
<a class="sourceLine" id="cb38-2" title="2">d &lt;-<span class="st"> </span><span class="kw">contamDE.lm</span>(simcounts,subtype)</a>
<a class="sourceLine" id="cb38-3" title="3"><span class="kw">names</span>(d)</a></code></pre></div>
<pre><code>## [1] &quot;counts&quot;        &quot;p.contamDE.lm&quot; &quot;log2FC&quot;        &quot;log2FC.cov&quot;   
## [5] &quot;proportion&quot;    &quot;design&quot;        &quot;df&quot;            &quot;weight&quot;       
## [9] &quot;y&quot;</code></pre>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb40-1" title="1"><span class="kw">head</span>(d<span class="op">$</span>p.contamDE.lm)</a></code></pre></div>
<pre><code>##      subtype-1 vs. normal subtype-2 vs. normal subtype-3 vs. normal
## [1,]          0.042134852           0.27356835           0.02203106
## [2,]          0.143430163           0.14678413           0.03125810
## [3,]          0.461327059           0.47160778           0.21565058
## [4,]          0.002098635           0.04750745           0.01110900
## [5,]          0.039278335           0.10545812           0.06870435
## [6,]          0.004599013           0.81536867           0.40552778</code></pre>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" title="1">d<span class="op">$</span>proportion</a></code></pre></div>
<pre><code>## Type1Tumor1 Type1Tumor2 Type1Tumor3 Type1Tumor4 Type2Tumor1 Type2Tumor2 
##   0.4067528   0.8861346   0.5380201   0.9639193   1.0000000   0.1542927 
## Type3Tumor1 Type3Tumor2 
##   0.6416266   0.9474743</code></pre>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb44-1" title="1"><span class="kw">head</span>(d<span class="op">$</span>log2FC)</a></code></pre></div>
<pre><code>##           [,1]       [,2]       [,3]
## [1,] 0.7864610  0.5589565  1.1594353
## [2,] 0.6035719  0.8956107  1.2305716
## [3,] 0.6549043  0.8796657 -1.3821024
## [4,] 2.1499433  1.6771908  1.9952982
## [5,] 0.8887638  0.9505932  0.9848645
## [6,] 2.6189108 -0.2359925  0.7463334</code></pre>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb46-1" title="1">d<span class="op">$</span>design</a></code></pre></div>
<pre><code>##   subtypeType1 subtypeType2 subtypeType3
## 1            1            0            0
## 2            1            0            0
## 3            1            0            0
## 4            1            0            0
## 5            0            1            0
## 6            0            1            0
## 7            0            0            1
## 8            0            0            1
## attr(,&quot;assign&quot;)
## [1] 1 1 1
## attr(,&quot;contrasts&quot;)
## attr(,&quot;contrasts&quot;)$subtype
## [1] &quot;contr.treatment&quot;</code></pre>
</div>
</div>
<div id="illustration-of-contamde.lm.test" class="section level1">
<h1>Illustration of <em>contamDE.lm.test</em></h1>

<p>Users should specify <em>d</em>, <em>H</em>,and <em>b</em> if they are interested in other DE analyses except DE analysis between tumor and normal samples. <em>d</em> is generally specified to be the list returned by the function ‘contamDE.lm’; <em>H</em> is a r x (p+q) design matrix of rank r and <em>b</em> is a p vector with the default value NULL (=0).</p>
<p>When it comes to the difference between any two or more subtypes (e.g. three tumor subtypes, subtype1 versus subtype2)</p>
<p>The <span class="math inline">\(H\)</span> under overall test of three tumor subtypes is</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb48-1" title="1">H &lt;-<span class="st"> </span><span class="kw">diag</span>(<span class="dv">3</span>)</a>
<a class="sourceLine" id="cb48-2" title="2">H</a></code></pre></div>
<pre><code>##      [,1] [,2] [,3]
## [1,]    1    0    0
## [2,]    0    1    0
## [3,]    0    0    1</code></pre>
<p>The <span class="math inline">\(H\)</span> under specific F test of subtype1 and subtype2 is</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb50-1" title="1">H &lt;-<span class="st"> </span><span class="kw">matrix</span>(<span class="kw">c</span>(<span class="dv">1</span>,<span class="op">-</span><span class="dv">1</span>,<span class="dv">0</span>),<span class="dv">1</span>,<span class="dv">3</span>);</a>
<a class="sourceLine" id="cb50-2" title="2">H</a></code></pre></div>
<pre><code>##      [,1] [,2] [,3]
## [1,]    1   -1    0</code></pre>
<p>Then the R code of conducting a F-test (overall test of three subtypes as an example) incorporating subtype information are</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb52-1" title="1"><span class="kw">library</span>(<span class="st">&quot;contamDE&quot;</span>)</a>
<a class="sourceLine" id="cb52-2" title="2">subtype &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="kw">rep</span>(<span class="st">&quot;Type1&quot;</span>,<span class="dv">4</span>),<span class="kw">rep</span>(<span class="st">&quot;Type2&quot;</span>,<span class="dv">2</span>),<span class="kw">rep</span>(<span class="st">&quot;Type3&quot;</span>,<span class="dv">2</span>))</a>
<a class="sourceLine" id="cb52-3" title="3">d &lt;-<span class="st"> </span><span class="kw">contamDE.lm</span>(simcounts,subtype)</a>
<a class="sourceLine" id="cb52-4" title="4">H &lt;-<span class="st"> </span><span class="kw">diag</span>(<span class="dv">3</span>)</a>
<a class="sourceLine" id="cb52-5" title="5">d1 &lt;-<span class="st"> </span><span class="kw">contamDE.lm.test</span>(d,H)</a></code></pre></div>
<p>The result is a list containing the following elements:</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb53-1" title="1"><span class="kw">names</span>(d1)</a></code></pre></div>
<pre><code>##  [1] &quot;counts&quot;             &quot;p.contamDE.lm&quot;      &quot;log2FC&quot;            
##  [4] &quot;log2FC.cov&quot;         &quot;proportion&quot;         &quot;design&quot;            
##  [7] &quot;df&quot;                 &quot;weight&quot;             &quot;y&quot;                 
## [10] &quot;p.contamDE.lm.test&quot;</code></pre>
<p>The respective p value of overall and specific F test by the new method are stored in</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb55-1" title="1"><span class="kw">head</span>(d1<span class="op">$</span>p.contamDE.lm.test)</a></code></pre></div>
<pre><code>## [1] 0.026225152 0.047547248 0.446028908 0.001783601 0.032940366 0.027460809</code></pre>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
